packages/augur-state/src/db/DB.ts

```diff
@@ -288,13 +294,79 @@ export class DB<TBigNumber> {
     return await this.syncableDatabases[dbName].find(request);
   }
 
+  /**
+   * Returns the current update_seqs from all SyncableDBs/UserSyncableDBs.
+   *
+   * TODO Remove this function if derived DBs are not used.
+   *
+   * @returns {Promise<SequenceIds>} Promise to a SequenceIds object
+   */
+  public async getAllSequenceIds(): Promise<SequenceIds> {
+    let sequenceIds: SequenceIds = {};
+    for (let eventName of this.genericEventNames) {
+      let dbName = this.getDatabaseName(eventName);
+      let dbInfo = await this.syncableDatabases[dbName].getInfo();
+      sequenceIds[dbName] = dbInfo.update_seq.toString();
+    }
+    for (let trackedUser of await this.trackedUsers.getUsers()) {
+      for (let userSpecificEvent of this.userSpecificEvents) {
+        let dbName = this.getDatabaseName(userSpecificEvent.name, trackedUser);
+        let dbInfo = await this.syncableDatabases[dbName].getInfo();
+        sequenceIds[dbName] = dbInfo.update_seq.toString();
+      }
+    }
+    return sequenceIds;
+  }
+
   /**
    * Queries the MetaDB.
    *
+   * TODO Remove this function if derived DBs are not used.
+   *
    * @param {PouchDB.Find.FindRequest<{}>} request Query object
    * @returns {Promise<PouchDB.Find.FindResponse<{}>>} Promise to a FindResponse
    */
   public async findInMetaDB(request: PouchDB.Find.FindRequest<{}>): Promise<PouchDB.Find.FindResponse<{}>> {
     return await this.metaDatabase.find(request);
   }
+
+  /**
+   * Saves raw logs from the blockchain to DBs, based on the event type for each log.
+   *
+   * @param {Provider} ethereumProvider
+   * @param {Events} events
+   * @param {Array<Log>} rawLogs
+   * @param {number} lastBlockNumberInChunk
+   */
+  private async saveLogsToDBs(ethereumProvider: Provider, events: Events, rawLogs: Array<Log>, lastBlockNumberInChunk: number): Promise<void> {
+    const saveLogsPromises = [];
+
+    // Save generic events to their respective DBs
+    for (let genericEventName of this.genericEventNames) {
+      const filteredLogs = rawLogs.filter((log) => log.topics.includes(ethereumProvider.getEventTopic("Augur", genericEventName)));
+      const parsedFilteredLogs = events.parseLogs(filteredLogs);
+      saveLogsPromises.push(this.getSyncableDatabase(genericEventName).saveLogs(parsedFilteredLogs, lastBlockNumberInChunk));
+      console.log(`Saved ${filteredLogs.length} ${genericEventName} log(s)`);
+    }
+
+    // Save user-specific events to their respective DBs
+    const trackedUsers = await this.trackedUsers.getUsers();
+    const encodedTrackedUsers = trackedUsers.map(trackedUser => `0x000000000000000000000000${trackedUser.substr(2)}`)
+    for (let userSpecificEvent of this.userSpecificEvents) {
+      const eventFilteredLogs = rawLogs.filter((log) => log.topics.includes(ethereumProvider.getEventTopic("Augur", userSpecificEvent.name)));
+        for (let trackedUserIndex = 0; trackedUserIndex < trackedUsers.length; trackedUserIndex++) {
```

Comment:
I recall tslint complaining about for(;;;) loops in favor of for (let i in list) which assigns I to the index.